modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js

```diff
@@ -498,6 +498,12 @@ AUI.add(
 					if (
 						currentTarget.hasClass('lfr-ddm-repeatable-add-button')
 					) {
+						var field = instance.getFieldDefinition();
+
+						if (field.type === 'select' && field.options[0].value === '') {
```

Comment:
Hey @pavel-savinov, you make a valid point there. I'm not sure what changed between when this got merged and now, but validation for options at the web structure page no longer works for checking the value is non-empty. So the first problem was only introduced recently, though right now I don't have the bandwidth to look into what the cause was. 

 For the second problem, I know that the current logic in master "shifts all repeatable fields". The changes in this PR should address that, because it's not using a forEach loop anymore, and instead directly accesses the field for which repeat is clicked through getFieldDefinition.

Shifting is important because we juggle the placeholder option between the backend and frontend, which causes a bug in indexing the Options array. What happens is the following:
1. ResourceCommand renderStructureField builds the Select field HTML with a placeholder ''.
2. In ddm_form.js, we have the original 3 options and need to manually add the placeholder '' to Options Array to mimic the HTML displayed.
3. We submit this form to renderStructureField upon repeating a field. Because the backend only expects "real" options in a Select field, another placeholder is added.
4. After repeating, instead of 4 options displayed, we now have a 5th which causes an array indexing error for getOptions.

Thus, the solution needed was to shift out the placeholder right before we repeat the Select Field. Since no other fields handle an extra value like Select, this edge case is contained to just this field.

In my opinion, option values should **not** be allowed to be empty, so we should make another ticket to handle that issue.